File: DSRFQ.Web/Modules/ServerTypes/Costing/CostingPartSpecialProcessResultsColumns.ts
import { ColumnsBase, fieldsProxy } from '@serenity-is/corelib';
import { Column } from '@serenity-is/sleekgrid';
import { CostingPartSpecialProcessResultsRow } from './CostingPartSpecialProcessResultsRow';

export interface CostingPartSpecialProcessResultsColumns {
    Id: Column<CostingPartSpecialProcessResultsRow>;
    CostingPartId: Column<CostingPartSpecialProcessResultsRow>;
    SpecialProcessName: Column<CostingPartSpecialProcessResultsRow>;
    SpecialProcessId: Column<CostingPartSpecialProcessResultsRow>;
    IsManual: Column<CostingPartSpecialProcessResultsRow>;
    InsertDate: Column<CostingPartSpecialProcessResultsRow>;
    InsertUserId: Column<CostingPartSpecialProcessResultsRow>;
    UpdateDate: Column<CostingPartSpecialProcessResultsRow>;
    UpdateUserId: Column<CostingPartSpecialProcessResultsRow>;
    DeleteDate: Column<CostingPartSpecialProcessResultsRow>;
    DeleteUserId: Column<CostingPartSpecialProcessResultsRow>;
    IsActive: Column<CostingPartSpecialProcessResultsRow>;
}

export class CostingPartSpecialProcessResultsColumns extends ColumnsBase<CostingPartSpecialProcessResultsRow> {
    static readonly columnsKey = 'Costing.CostingPartSpecialProcessResults';
    static readonly Fields = fieldsProxy<CostingPartSpecialProcessResultsColumns>();
}

File: DSRFQ.Web/Modules/Costing/CostingPartSpecialProcessResults/CostingPartSpecialProcessResultsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace DSRFQ.Costing.Columns;

[ColumnsScript("Costing.CostingPartSpecialProcessResults")]
[BasedOnRow(typeof(CostingPartSpecialProcessResultsRow), CheckNames = true)]
public class CostingPartSpecialProcessResultsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string CostingPartPartNumber { get; set; }
    [EditLink]
    public string SpecialProcessName { get; set; }
    public int SpecialProcessId { get; set; }
    public int IsManual { get; set; }
    public DateTime InsertDate { get; set; }
    public string InsertBy { get; set; }
    public DateTime UpdateDate { get; set; }
    public string UpdateBy { get; set; }
    public DateTime DeleteDate { get; set; }
    public string DeleteBy { get; set; }
}